components/log_archive/main.tf

```diff
@@ -0,0 +1,27 @@
+variable "name" {
+  description = "The name for the archive bucket (e.g. dosomething-papertrail)."
+}
+
+resource "aws_s3_bucket" "archive" {
+  bucket = var.name
+  acl    = "private"
+
+  tags = {
+    Application = "papertrail"
+    Environment = "all"
+    Stack       = "all"
+  }
+}
+
+resource "aws_s3_bucket_policy" "papertrail_access_policy" {
+  bucket = aws_s3_bucket.archive.id
+  policy = templatefile("${path.module}/policy.json.tpl", { bucket_arn = aws_s3_bucket.archive.arn })
```

Comment:
So this line and the `policy.json.tpl` file are saying that the Papertrail AWS user can put and delete things in this bucket?

Reply:
Yep!